### ISSUE TITLE
AppMetrics missing status codes

### ISSUE THREAD

**Issue by ns3777k** [artprima/prometheus-metrics-bundle#6]
Hey! I've a couple of questions:

1. Is it possible to turn off the bundled AppMetrics without traversing services and removing it? Like an options or something? I couldn't find one.
2. Any reason why status codes are missing? I see 2xx, 4xx, etc. But a common case is to differentiate 404 from 403 (in grafana graphs for example). Should I submit a PR or add them somehow in my project?

**Comment by denisvmedia** [comment]
1. No, there is no way to turn it off. But yes, it would be nice to add a feature to turn it off. But I'd prefer to keep it on by default, so that the bundle provides some metrics right after the installation. Do you have any suggestion on how to make it pluggable without traversing the services? You can actually, override the service itself by your own class, but I'm not sure if this is an acceptable solution.
2. There is no particular reason, except that I didn't want to have a metric per status code (yes, distinguishing between 4xx codes could be important, but it's not that relevant for the other code types). I actually consider AppMetrics as an example. And as such, if we find a convenient way how to allow turning it off, this would let one create his own metrics instead of the default ones.

**Comment by ns3777k** [comment]
Well, how about registering `AppMetrics` just like every custom metrics generator instead of doing this in bundle's `services.xml`?

```yaml
Artprima\PrometheusMetricsBundle\Metrics\AppMetrics:
        tags:
            - { name: prometheus_metrics_bundle.metrics_generator }
```

Adding this in flex recipe will let it be on automatically.

**Comment by ns3777k** [comment]
@denisvmedia what do you think ? :-)

**Comment by ns3777k** [comment]
ok :-) thx

**Comment by denisvmedia** [comment]
Ok, so... Let's divide it to two steps. First of all, I agree to remove the service by default. Also, I agree to include that in readme. This would be the first step. As soon as we done it, we can release a new version. After that, we can make a recipe that would get back the defaults. Why would we do that split into two steps? Simply because I think it is only possible to submit the recipes only after you have a version released, otherwise they will decline a pull request.

The only problem, however, is that I don't have time to do the stuff (I can make a pull request for the recipe, but that's it). Would you be eager to provide a pull request to the prometheus metrics bundle repo according to the agreed conditions? I would then release both of your changes so that we could speed up the process.

**Comment by denisvmedia** [comment]
cc @ns3777k

**Comment by ns3777k** [comment]
Hey! Actually I found out that this bundle has more disadvantages for me than I thought, so I wrote another one for my project. Things I needed:

1. Is metricsGenerator is a really need feature? Most of the time you use only one to record request and response metrics.
2. No information on how to create metrics not related to request / response
3. Injecting CollectorRegistry into services requires to also inject namespace

I created only 1 listener to collect metrics from request and response and made a decorator service to wrap CollectorRegistry so I always have a namespace injected.
